### ISSUE TITLE
Update datasource and cell value

### ISSUE THREAD

**Issue by stavrianosy** [fooplugins/FooTable#501]
The footable is populated with a list of messages which are fetched with an ajax call. 
One of the column is using the formatter to display an envelope icon to show if the message is read/unread. 

I need allow the user to click on the envelope icon and change the status from read to unread. 
Later when the user read all messages there will be a button to submit the changes or even the whole datasource list back on the server.

`$('.myTable').footable({
            "rows": this.ftResult,
            "paging": {"limit": 10,"enabled":true, "size":15, "current":1, "position": "right"},
            "filtering": {
                "enabled":true,
            }
        });`

How can I achieve this updating on the cell value/icon and the datasource?

Thanks for the great plugin.

**Comment by steveush** [comment]
Hi,

I would bind a delegate click event on the table itself and then in the handler simply toggle the value. As a simple example which should get you headed in the right direction:

``` javascript
$('.myFooTable').on('click', '.read-unread-icon', function(e){
    e.preventDefault();
    // to get the FooTable.Row object simply walk up the DOM, find the TR and grab the __FooTableRow__ data value
    var $row = $(this).closest('tr');
    // if this is a detail row get the previous row in the table to get the main TR element
    if ($row.hasClass('footable-detail-row')){
        $row = $row.prev();
    }
    // grab the row object
    var row = $row.data('__FooTableRow__');
    // get the cell you want to work with, you'll need to figure out the index you need to use
    var cell = row.cells[0];
    // you can then use the FooTable.Cell.val() function to get or set the cell value (it updates the DOM using the formatter)
    var current = cell.val();
    if (current === 'read'){
        cell.val('unread');
    } else {
        cell.val('read');
    }
});
```

Things you would obviously need to change in the above example would be:
1. The delegate selector `'.read-unread-icon'` must be changed to match whatever your icon is.
2. The index of the cell needs to be changed to match the correct column in the table. Either hard code it as above or simply loop through the `row.cells` array checking for the name or something.
3. Change the `if else` check to match your actual column values.

NOTE: As you said you are using a custom formatter to create the icon please note the value you supply to the `cell.val()` function gets passed to your formatter to create the cell contents.

Thanks

**Comment by stavrianosy** [comment]
Thanks for the prompt response

I get an error at the 
`var row = $row.data('__FooTableRow__');`
`var cell = row.cells[0];`
Cannot read property 'cells' of undefined

Do i need to change the name of **FooTableRow** to something else ?

**Comment by steveush** [comment]
Hi,

No you shouldn't have to, the above is the exact code I used locally to test with and it worked. I've actually created a new [helper function](https://github.com/fooplugins/FooTable/blob/V3/src/js/FooTable.js#L75) which uses the code to retrieve the row object. The helper is not in a live version yet and will only be available in 3.0.12 unless you want to check out the repo and build the project locally, I was just mentioning it so you know the code works ;)

Any chance you can create a simplified jsfiddle or provide a url showing the issue? There really should be a `FooTable.Row` object when you call `$row.data('__FooTableRow__')`. We've used this method to create custom toggles and the like previously and without being able to see you table structure or debug the code I can't say why it is not working for you at the moment.

Thanks

**Comment by lmatra** [comment]
I am also seeing this problem from stavrianosy on August 18, 2016 and I'm running version 3.1.5.  Is there a new way to retrieve the value of a cell?
